pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -0,0 +1,256 @@
+)
+
+// TODO: Register the capability in the NewDelegateFunction in chainlink/core/services/workflows.delegate.go
+
+var mercuryInfo = capabilities.MustNewCapabilityInfo(
+	"mercury-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example mercury trigger.",
+	"v1.0.0",
+)
+
+// This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
+type MercuryTriggerService struct {
+	capabilities.CapabilityInfo
+	chans                 map[string]chan<- capabilities.CapabilityResponse
+	feedIDsForTriggerID   map[string][]int64 // TODO: switch this to uint64 when value.go supports it
+	triggerIDsForFeedID   map[int64][]string
+	mu                    sync.Mutex
+	triggerIDToWorkflowID map[string]string
+}
+
+var _ capabilities.TriggerCapability = (*MercuryTriggerService)(nil)
+
+func NewMercuryTriggerService() *MercuryTriggerService {
+	return &MercuryTriggerService{
+		CapabilityInfo:        mercuryInfo,
+		chans:                 map[string]chan<- capabilities.CapabilityResponse{},
+		feedIDsForTriggerID:   make(map[string][]int64),
+		triggerIDsForFeedID:   make(map[int64][]string),
+		triggerIDToWorkflowID: make(map[string]string),
+	}
+}
+
+// maybe use mercury-pipline reports.go report struct instead of MercuryReport struct, and then we can use the pipeline report generators
+func (o *MercuryTriggerService) ProcessReport(reports []mercury.FeedReport) error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	currentTime := time.Now()
+	unixTimestampMillis := currentTime.UnixNano() / int64(time.Millisecond)
+	triggerIDsForReports := make(map[string][]int)
+	reportIndex := 0
+
+	for _, report := range reports {
+		for _, triggerID := range o.triggerIDsForFeedID[report.FeedID] {
+			// if its not initialized, initialize it
+			if _, ok := triggerIDsForReports[triggerID]; !ok {
+				triggerIDsForReports[triggerID] = make([]int, 0)
+			}
+			triggerIDsForReports[triggerID] = append(triggerIDsForReports[triggerID], reportIndex)
+		}
+		reportIndex++
+	}
+
+	// Then for each trigger id, find which reports correspond to that trigger and create an event bundling the reports
+	// and send it to the channel associated with the trigger id.
+	for triggerID, reportIDs := range triggerIDsForReports {
+		reportPayload := make([]mercury.FeedReport, 0)
+		for _, reportID := range reportIDs {
+			reportPayload = append(reportPayload, reports[reportID])
+		}
+
+		triggerEvent := mercury.TriggerEvent{
+			TriggerType: "mercury",
+			ID:          GenerateTriggerEventID(reportPayload),
+			Timestamp:   strconv.FormatInt(unixTimestampMillis, 10),
+			Payload:     reportPayload,
+		}
+
+		val, err := mercury.Codec{}.WrapMercuryTriggerEvent(triggerEvent)
+		if err != nil {
+			return err
+		}
+
+		// Create a new CapabilityResponse with the MercuryTriggerEvent
+		capabilityResponse := capabilities.CapabilityResponse{
+			Value: val,
+		}
+
+		wfID := o.triggerIDToWorkflowID[triggerID]
+		ch, ok := o.chans[wfID+triggerID]
```

Comment:
Extract the logic of building an internal key (wfID+triggerID) to a helper functions